app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -159,59 +206,64 @@ class ConversationListViewModel @Inject constructor(
+        filter { it is Group || it is OneOne }
+            .map {
+                it.toItem(
+                    private = { privateConversation -> GeneralConversation(privateConversation) },
+                    group = { groupConversation -> GeneralConversation(groupConversation) },
                 )
             }
-            is OneOne -> {
-                val otherUser = details.otherUser
-
-                GeneralConversation(
-                    ConversationType.PrivateConversation(
-                        userInfo = UserInfo(
-                            otherUser.previewPicture?.let { UserAvatarAsset(it) },
-                            UserStatus.NONE // TODO Get actual status
-                        ),
-                        conversationInfo = ConversationInfo(
-                            name = otherUser.name.orEmpty(),
-                            membership = mapUserType(details.userType)
-                        ),
-                        conversationId = conversation.id,
-                        mutedStatus = conversation.mutedStatus,
-                        isLegalHold = details.legalHoldStatus.showLegalHoldIndicator()
-                    )
-                )
-            }
-            is Self -> {
-                throw IllegalArgumentException("Self conversations should not be visible to the user.")
-            }
-        }
+
+}
+
+private fun LegalHoldStatus.showLegalHoldIndicator() = this == LegalHoldStatus.ENABLED
+
+private fun ConversationDetails.toItem(
+    private: (ConversationType.PrivateConversation) -> ConversationItem,
+    group: (ConversationType.GroupConversation) -> ConversationItem,
+): ConversationItem = when (this) {
+    is Group -> {
+        group(
+            ConversationType.GroupConversation(
+                groupColorValue = getConversationColor(conversation.id),
+                groupName = conversation.name.orEmpty(),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator()
+            )
+        )
+    }
+    is OneOne -> {
+        private(
+            ConversationType.PrivateConversation(
+                userInfo = UserInfo(
+                    otherUser.previewPicture?.let { UserAvatarAsset(it) },
+                    UserStatus.NONE // TODO Get actual status
+                ),
+                conversationInfo = ConversationInfo(
+                    name = otherUser.name.orEmpty(),
+                    membership = userType.toMembership()
+                ),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator()
+            )
+        )
+    }
+    is Self -> {
+        throw IllegalArgumentException("Self conversations should not be visible to the user.")
     }
+}
```

Comment:
like the idea ! :) something I ahve not seen before but, not sure if this function is needed : this would be sufficent I think ? 

```
when (it) {
    is Group -> {
            ConversationType.GroupConversation(
                groupColorValue = getConversationColor(conversation.id),
                groupName = conversation.name.orEmpty(),
                conversationId = conversation.id,
                mutedStatus = conversation.mutedStatus,
                isLegalHold = legalHoldStatus.showLegalHoldIndicator()
            )
        )
    }
    is OneOne -> {
 
            ConversationType.PrivateConversation(
                userInfo = UserInfo(
                    otherUser.previewPicture?.let { UserAvatarAsset(it) },
                    UserStatus.NONE // TODO Get actual status
                ),
                conversationInfo = ConversationInfo(
                    name = otherUser.name.orEmpty(),
                    membership = userType.toMembership()
                ),
                conversationId = conversation.id,
                mutedStatus = conversation.mutedStatus,
                isLegalHold = legalHoldStatus.showLegalHoldIndicator()
            )
       
    }
    is Self -> {
        throw IllegalArgumentException("Self conversations should not be visible to the user.")
    }
```

Reply:
I need to return `ConversationItem` not `ConversationType` :D